adobe/helix-importer
# feat: add support for base64 images

### Context

Contribution following code sample shared with customer to solve the issue on a specific project.


### What it does

* Add support for Base64 images in `preprocess` function. 
  Then, base64 images were removed 
  Now, original base64 images are modified: the base64 inline data is replaced by an object URL (generated using https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL_static)
* Add Unit tests


### How to test

1. Clone https://github.com/hlxsites/petplace
2. Checkout branch `issue-28`
3. From the root project, in a terminal, run `hlx import` to fetch latest `helix-importer-ui` then stop it already.
4. From `tools/importer/helix-importer-ui/`:
   * in `package.json` modify helix-importer dependency to `adobe/helix-importer#feat-support-b64-images`
   * `rm node_modules`
   * `npm install`
   * `npm run build:dev`
5. Re-start Helix Importer server without fetching `helix-importer-ui` project, `hlx import --skip-ui`
6. In Helix Importer UI:
   * use test URL `https://www.petplace.com/article/breed/pembroke-welsh-corgi/`
   * use importer script `http://localhost:3001/tools/importer/import.js`
   * run import
7. In Preview tab, confirm that you see following images in the "Care Tabs" block:
   ![Screenshot 2023-05-23 at 15 46 48](https://github.com/adobe/helix-importer/assets/40354404/e3723cad-fdcc-468d-af57-23fc4a6f7d6b)
